var/spack/repos/builtin/packages/tcptrace/package.py

```diff
@@ -0,0 +1,57 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+from os.path import join
+
+
+class Tcptrace(AutotoolsPackage):
+    """tcptrace is a tool written by Shawn Ostermann at Ohio University for
+       analysis of TCP dump files. It can take as input the files produced by
+       several popular packet-capture programs, including tcpdump, snoop,
+       etherpeek, HP Net Metrix, and WinDump."""
+
+    homepage = "http://www.tcptrace.org/"
+    url      = "http://www.tcptrace.org/download/tcptrace-6.6.7.tar.gz"
+
+    version('6.6.7', '68128dc1817b866475e2f048e158f5b9')
+
+    depends_on('bison', type='build')
+    depends_on('flex', type='build')
+    depends_on('libpcap')
+
+    patch('tcpdump.patch')
```

Comment:
@sknigh did this patch come from somewhere? If so IMO it would be worthwhile to add the link here (and a summary of what the patch does). If you wrote it yourself it would also be worth knowing why it was needed.

Reply:
It was derived from [this rejected PR](https://github.com/blitz/tcptrace/pull/2), which has a broken URL to (what I assume was the same issue in) the Arch Linux package repository. 

Before the patch, I was running into this error
```
tcpdump.c:(.text+0x34): undefined reference to `pcap_offline_read
```

The error comes from a change in how libpcap API exports symbols, which was [reported to Debian maintainers in 2009](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=545595), but tcptrace is not actively maintained any more and the fix for Arch and Debian has not merged into the upstream source. I can add a link to the Debian message log when I get the chance.